### PR TITLE
python: mypy version bump

### PR DESCRIPTION
**Pull request by phlogistonjohn**
<s>NOTE: This is a draft initially because, despited being "done", I only have tried it locally. I know the ci env is quite different from mine. Once I'm certain this doesn't interfere with normal make check (and other) jobs in the CI I'll move it out of draft.</s>

In PR #55255 I updated various files to be valid on newer versions of
mypy. Update the constrains file to specify mypy 1.1.1. This certainly
is not the newest version of mypy, but because ceph uses various older
versions of python I am being cautious and trying to only incrementally
update the mypy version. This version fixed an issue or two I hit while
working on a prototype manager module compared to the previous version
in the mypy-constrains.txt file.

In addition, I am cleaning some small mypy and related annoyances I bumped into while working on a new mgr module in the src/pybind/mgr tox.ini file.

Lastly, there's a commit in here that's a duplicate of one in #55068 . I did this because the python versions on my local machine default to versions that fail without the change. Depending on which PR is merged first I will end up dropping it from the other.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
  - [x] Testing tool configuration updates
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
  - [x] I'm open to suggestions
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate. I'm confident on this one!
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests
  - [x] Tweaks the testing environments

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
Failure:
```
      self = <snap_schedule.tests.fs.test_schedule.TestSchedule object at 0x7f50f2d200d0>
simple_schedule = <snap_schedule.fs.schedule.Schedule object at 0x7f50bc379900>

    def test_created_now(self, simple_schedule):
        now = datetime.datetime.now(datetime.timezone.utc)
>       assert simple_schedule.created.minute == now.minute
E       assert 18 == 19
E        +  where 18 = datetime.datetime(2024, 3, 18, 21, 18, 59, 999804, tzinfo=datetime.timezone.utc).minute
E        +    where datetime.datetime(2024, 3, 18, 21, 18, 59, 999804, tzinfo=datetime.timezone.utc) = <snap_schedule.fs.schedule.Schedule object at 0x7f50bc379900>.created
E        +  and   19 = datetime.datetime(2024, 3, 18, 21, 19, 0, 299, tzinfo=datetime.timezone.utc).minute

snap_schedule/tests/fs/test_schedule.py:65: AssertionError
```

Looks unrelated to my changes.



```      
      mypy: OK ✔ in 1 minute 46.71 seconds
```

Mypy was ok.


Will rerun make check later to see if the failure was transient.

**Comment by phlogistonjohn** [comment]
jenkins test make check

**Comment by phlogistonjohn** [comment]
jenkins test api

File: src/pybind/mgr/tox.ini [hunk 1]
```diff
@@ -65,12 +65,11 @@ commands =
     pytest {posargs:cephadm/tests/test_ssh.py}
 
 
-[testenv:mypy]
+[testenv:{,py37-,py38-,py39-,py310-}mypy]
```
Comment (adk3798) [inline src/pybind/mgr/tox.ini:68]: to be clear, this gives us the ability to run something like `tox -e py37-mypy`, but by default we'll still just run `mypy`?

**Review by adk3798 — APPROVED** [review]

File: src/pybind/mgr/tox.ini (hunk 1 above)
Comment (phlogistonjohn) [reply src/pybind/mgr/tox.ini:68]: Yep. tox supports expansions like those of bash. So `{,py37-,py38-,py39-,py310-}mypy` expands to mypy, py37-mypy, py39-mypy, py310-mypy.  I got lazy and didn't figure we'd need anything newer than 3.10 for now.

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-26_01:50:53-orch:cephadm-wip-adk-testing-2024-03-25-1628-distro-default-smithi/

Other than the `in cluster log` stuff, 4 failures were the mds upgrade sequence which is known to fail currently, and the test_cephadm task, which failed with https://tracker.ceph.com/issues/65155

Nothing to block merging the PRs in the run

**Comment by adk3798** [comment]
jenkins test api

**Comment by phlogistonjohn** [comment]
jenkins test api

**Pull request merged**